packages/browser/test/e2e/test/breadcrumbs.test.js

```diff
@@ -0,0 +1,604 @@
+const { test, expect } = require('@playwright/test');
+const Sentry = require('@sentry/browser');
+
+const { waitForXHR, getSentryRequest } = require('./utils/helpers');
+
+test.describe('Breadcrumbs', () => {
+  test.beforeEach(async ({ baseURL, page }) => {
+    await page.goto(baseURL);
+  });
+
+  test('should record an XMLHttpRequest with a handler', async ({ page }) => {
+    const eventData = await getSentryRequest(
+      page,
+      () => {
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', '/base/subjects/example.json');
+        xhr.onreadystatechange = function() {};
+        xhr.send();
+
+        waitForXHR(xhr, function() {
+          Sentry.captureMessage('test');
+        });
+      },
+      { waitForXHR },
+    );
+
+    expect(eventData.message).toBe('test');
+    expect(eventData.breadcrumbs.length).toBe(1);
+    expect(eventData.breadcrumbs[0].category).toBe('xhr');
+    expect(eventData.breadcrumbs[0].type).toBe('http');
+    expect(eventData.breadcrumbs[0].data.method).toBe('GET');
+    expect(eventData.breadcrumbs[0].data.url).toBe('/base/subjects/example.json');
+  });
+
+  test('should record an XMLHttpRequest with a handler attached after send was called', async ({ page }) => {
+    const eventData = await getSentryRequest(
+      page,
+      () => {
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', '/base/subjects/example.json');
+        xhr.onreadystatechange = function() {
+          window.handlerCalled = true;
+        };
+        xhr.send();
+
+        waitForXHR(xhr, function() {
+          Sentry.captureMessage('test');
+        });
+      },
+      { waitForXHR },
+    );
+
+    const handlerCalled = await (await page.evaluateHandle(() => Promise.resolve(window.handlerCalled))).jsonValue();
+
+    expect(handlerCalled).toBe(true);
+    expect(eventData.message).toBe('test');
+    expect(eventData.breadcrumbs.length).toBe(1);
+    expect(eventData.breadcrumbs[0].category).toBe('xhr');
+    expect(eventData.breadcrumbs[0].type).toBe('http');
```

Comment:
I wonder if it'll be effective to us to pseudo snapshot test the event instead of these deep assertions.

So we can skip attributes that we know will change, but compare the rest of the event to a json file.

We do something similar in electron:

An event JSON: https://github.com/getsentry/sentry-electron/blob/master/test/e2e/test-apps/javascript/main-error/event.json

and how it's normalized: https://github.com/getsentry/sentry-electron/blob/master/test/e2e/recipe/normalize.ts#L49

We can also try this out after this gets merged in, no rush